moesif-servlet/src/main/java/com/moesif/servlet/BatchProcessor.java

```diff
@@ -0,0 +1,150 @@
+
+    private final MoesifConfiguration moesifConfig;
+    private final MoesifAPIClient moesifApi;
+    private final Boolean debug;
+    private final BlockingQueue<EventModel> batchQueue;    // Queue to hold the batched events
+
+    private boolean jobRunning = false;
+
+    public BatchProcessor(MoesifAPIClient moesifApi, MoesifConfiguration moesifConfig, boolean debug) {
+        this.moesifApi = moesifApi;
+        this.moesifConfig = moesifConfig;
+        this.debug = debug;
+
+        // Initialize the queue with queue size capacity.
+        this.batchQueue = new ArrayBlockingQueue<>(this.moesifConfig.eventQueueSize);
+    }
+
+    public int getBatchMaxTimeInSec() {
+        return this.moesifConfig.batchMaxTimeInSec;
+    }
+
+    public int getUpdateConfigTimeInMin() {
+        return this.moesifConfig.updateConfigTimeInMin;
+    }
+
+    /***
+     * Method to add event into a queue for batch-based event transfer. It maintains
+     * time when the last event was added to the event queue.
+     * @param event: EventModel
+     */
+    public void addEvent(EventModel event) {
+        try {
+            // Add the event to the queue
+            this.batchQueue.put(event);
+
+            if (this.debug) {
+                String msg = String.format("Event successfully added to queue. Queue Size = %d", this.batchQueue.size());
+                logger.info(msg);
+            }
+
+        } catch (InterruptedException e) {
+            logger.warning("Add event failed. " + e.toString());
+        } catch (Exception e) {
+            logger.warning("Add event failed. " + e.toString());
+        }
+
+    }
+
+    /***
+     * Method to check if job is running or not.
+     * @return boolean
+     *   Returns true if job is running otherwise false.
+     */
+    public boolean isJobRunning() {
+        return this.jobRunning;
+    }
+
+    /***
+     * Task runner method to send events in batches if at least a batch of events available.
+     * It also keeps track of a flag to show if job is in-progress.
+     * If debug has been enabled then it prints how many batches were sent with total events count.
+     */
+    @Override
+    public void run() {
+        // set jobRunning
+        this.jobRunning = true;
+
+        // Return if queue is empty
+        if (this.batchQueue.isEmpty()) {
+            if (this.debug) {
+                logger.info("No events to send.");
+            }
+
+            // reset jobRunning
+            this.jobRunning = false;
+            return;
+        }
+
+        // callback for async createBatchEvents
+        APICallBack<com.moesif.api.http.response.HttpResponse> callBack = new APICallBack<com.moesif.api.http.response.HttpResponse>() {
+
+            @Override
+            public void onSuccess(HttpContext httpContext, HttpResponse httpResponse) {
+                // ignore
+            }
+
+            public void onFailure(HttpContext context, Throwable error) {
+                // ignore
```

Comment:
im not too familiar with this but, for `onSuccess` it may still have status code that is 4xx ot 5xx? [See sdk link](https://github.com/Moesif/moesifapi-java#2a-send-the-events-batch-asynchronously) and may need to treat them as failure? What would be the behavior with token that fails auth all the time vs intermittent network failure?
```
assertEquals("Status is not 201",
                201, httpContext.getResponse().getStatusCode());
```

Reply:
Done